### PR TITLE
change pipeline version threshold for assembly

### PR DESCRIPTION
**Pull request by cdebourcy**
Give ourselves more time!

### CODE REVIEW

File: app/models/pipeline_run.rb [hunk 1]
```diff
@@ -402,7 +402,7 @@ def multihit?
   end
 
   def assembly?
-    after(pipeline_version || fetch_pipeline_version, "1.11")
+    after(pipeline_version || fetch_pipeline_version, "1000.1000")
```
Comment (yunfangjuan) [inline app/models/pipeline_run.rb]: which means no assembly?

Comment (cdebourcy) [reply app/models/pipeline_run.rb]: yes. We decided to hold off on pushing the assembly pipeline (or other new features) until after the refactor.

Comment (yunfangjuan) [reply app/models/pipeline_run.rb]: put some documentation on it to indicate the intention. This code looks cryptic by itself.

Comment (cdebourcy) [reply app/models/pipeline_run.rb]: ok!

**Comment by yunfangjuan** [comment]
```
<?php

// Input is FASTA, paire-end reads
// USAGE:  -1 [read1.fasta] -2 [read2.fasta] -c [0.38] -t


class LZW
{
    function compress($unc) {
        $i;$c;$wc;
        $w = "";
        $dictionary = array();
        $result = array();
        $dictSize = 256;
        for ($i = 0; $i < 256; $i += 1) {
            $dictionary[chr($i)] = $i;
        }
        for ($i = 0; $i < strlen($unc); $i++) {
            $c = $unc[$i];
            $wc = $w.$c;
            if (array_key_exists($w.$c, $dictionary)) {
                $w = $w.$c;
            } else {
                array_push($result,$dictionary[$w]);
                $dictionary[$wc] = $dictSize++;
                $w = (string)$c;
            }
        }
        if ($w !== "") {
            array_push($result,$dictionary[$w]);
        }
        $fraction = (count($result) / strlen($unc) );
        return ($fraction);
    }
}


// y = -0.0009x + StartingFraction
// where x is nt, starting at 100nt

// x = 0.468 at 100nt
// x = 0.437 at 135nt

// if x<135, then fraction = fraction - ((135-x)*0.0009)
// only important for stuff with initial fraction of <0.42

// for x>135, trend is back down, so need to add correction to fraction

$options = getopt("1:2:c:th");


if (key_exists("h",$options)) {
	print "USAGE:  -1 [read1.fasta] -2 [read2.fasta] -c [0.38] -t\n";
	print "-t writes file of reads that do not pass cutoff\n";
	print "-h help\n";
	die;

}

if (empty($options["1"]) || empty($options[2])) {
	print "USAGE:  -1 [read1.fasta] -2 [read2.fasta] -c [0.38] -t\n";
	die;
	}

$basename1 = $options["1"];
$basename2 = $options["2"];

$outfile1 = "lzw.".$basename1;
$outfile2 = "lzw.".$basename2;

$cutoff = floatval($options["c"]);

if (key_exists("t",$options)) {
	$trash = true;
	}
else { $trash = false; }


if (empty($cutoff)) { $cutoff = 0.38; }


$keep =0; $discard=0;

$source_file1 = fopen($basename1,'r','1') or die("Error. Couldn't open local file: $basename1\n");
$source_file2 = fopen($basename2,'r','1') or die("Error. Couldn't open local file: $basename2\n");


$dest_file1  = fopen($outfile1,'w') or die("Error. Couldn't open local dest file: $outfile1\n");
$dest_file2  = fopen($outfile2,'w') or die("Error. Couldn't open local dest file: $outfile2\n");
$dest_trash  = fopen("lzw.trash",'w') or die("Error. Couldn't open local dest file: trash\n");

$lzw = new LZW();


// MAIN

$count =0;
$keep = 0;
$discard = 0;
$total = 0;

while (!feof($source_file1)) {


  // FASTA FILE - GET 2 lines from each of the paired ends.


	$line_r1_1 = fgets($source_file1);
	$line_r1_2 = fgets($source_file1);
	$line_r2_1 = fgets($source_file2);
	$line_r2_2 = fgets($source_file2);

    if ($count == 0) {
    	 if ($line_r1_1[0] != ">") {
    	 	print "Not a fasta?\n"; die; }
    	 $length = strlen($line_r1_2);
    	 $correction = (135-$length)*0.0009;
	}
    if (strlen($line_r1_2) > 25) {
		$fraction1 = ($lzw->compress($line_r1_2)) - $correction;
		$fraction2 = ($lzw->compress($line_r2_2)) - $correction;
		$total = $total + $fraction1 + $fraction2;
	}


	if (($fraction1 > $cutoff) && ($fraction2 > $cutoff)) {
		$keep++;

		fwrite($dest_file1,$line_r1_1);
		fwrite($dest_file1,$line_r1_2);
		fwrite($dest_file2,$line_r2_1);
		fwrite($dest_file2,$line_r2_2);

		}
	else {
		$discard++;
		if ($trash) {
			if ($fraction1 < $cutoff) {
				fwrite($dest_trash,$line_r1_2);
				}
			if ($fraction2 < $cutoff) {
				fwrite($dest_trash,$line_r2_2);
				}
		  }
		}

	$count ++;


}

$avg = sprintf("%2.2f",( ($total) / (2*$count))*100);


if ($keep>0) {
		$final = sprintf("%2.2f",($keep / $count)*100);
		print "$count\t$avg\t$keep\t$discard\t$final\n";
		}
	else {  print "$count - Nothing kept.\n";}

fclose($source_file1);
fclose($source_file2);
fclose($dest_file1);
fclose($dest_file2);

exit(0);
```

**Pull request merged**